drivers/SmartThings/matter-switch/src/eve-energy/init.lua

```diff
@@ -41,7 +42,9 @@ local REPORT_TIMEOUT = (15 * 60) -- Report the value each 15 minutes
 -------------------------------------------------------------------------------------
 
 local function is_eve_energy_products(opts, driver, device)
-  if device.manufacturer_info.vendor_id == EVE_MANUFACTURER_ID then
+  -- this sub driver does not support child devices
+  if device.network_type == device_lib.NETWORK_TYPE_MATTER and
+     device.manufacturer_info.vendor_id == EVE_MANUFACTURER_ID then
```

Comment:
This will often require 2 operations, whereas if the eve mfr check was first it will often only need 1.

Reply:
`if device.network_type == device_lib.NETWORK_TYPE_MATTER` is put first because child devices do not have `manufacturer_info`, so calling `can_handle` for the child devices results in a warning. Therefore, I added the check for the `network_type` first to prevent the warning.